src/manifold/src/manifold.cpp

```diff
@@ -881,4 +882,73 @@ Manifold Manifold::Hull() const { return Hull(GetMesh().vertPos); }
 Manifold Manifold::Hull(const std::vector<Manifold>& manifolds) {
   return Compose(manifolds).Hull();
 }
+
+/**
+ * Computes the minimum distance between two manifolds. Returns a float between
+ * 0 and searchLength.
+ *
+ * @param other The other manifold to compute the minimum gap to.
+ * @param searchLength The maximum distance to search for a minimum gap.
+ */
+float Manifold::MinGap(const Manifold& other, float searchLength) const {
+  auto intersect = *this ^ other;
+  auto prop = intersect.GetProperties();
+
+  if (prop.volume != 0) return 0.0f;
+
+  auto getSortedExpandedFaceBoxMorton = [searchLength](
+                                            const Manifold& manifold) {
+    Vec<Box> faceBox;
+    Vec<uint32_t> faceMorton;
+
+    manifold.GetCsgLeafNode().GetImpl()->GetFaceBoxMorton(faceBox, faceMorton);
+
+    transform(autoPolicy(faceBox.size()), faceBox.begin(), faceBox.end(),
+              faceBox.begin(), [searchLength](const Box& box) {
+                return Box(box.min - glm::vec3(searchLength),
+                           box.max + glm::vec3(searchLength));
+              });
+
+    manifold.GetCsgLeafNode().GetImpl()->SortFaceBoxMorton(faceBox, faceMorton);
```

Comment:
Since `SortFaceBoxMorton` only seems to be called once in the whole library, I think I'd prefer to just inline it here instead of making a new `Impl` member.

Reply:
You may be right, do we need to sort here at all? The Morton codes are based on the centers of the bounding boxes and we are just expanding them here, which should leave their centres untouched. The tests pass without the sorting as well

Reply:
Please don't take my word for it - have a look at `GetFaceBoxMorton` and make sure. You can test this by rotating a sphere - I have the feeling your newly calculated Morton codes won't match the sorting that was done before rotation.

Reply:
I think it should be fine if you check against different test cases. I can run some larger tests with openscad later, when everything else is addressed. Please ping me at that point, I am a bit busy recently and may not respond quickly. (grading assignments...)

Reply:
Good to know it's correct - but the point of the BVH is speed. To ensure it hasn't messed up the tree, I'd take a very high-res sphere and check the rotated against the non-rotated for timing. If they're pretty similar, then it's probably fine.

Reply:
Looks like we might need to sort after all. If I check the min gap between two rotated spheres without sorting, I get a Vec out of range exception.